src/expression/logical/and.ts

```diff
@@ -37,4 +38,29 @@ export class And extends Logical {
     }
     return true
   }
+
+  simplify (ctx: Context): boolean | Evaluable {
```

Comment:
Why would `simplify` return `boolean`?

Reply:
I think I get it now.  I find it strange when a method can return more than one type. I've seen this done a tuple:
`[boolean, Evaluable]`

Reply:
In this case it is not a tuple. In functional programming this would be an either. And if were a tuple it would need to be `[boolean | undefined, Evaluable | undefined]`

Reply:
I do not believe so that the intention here is to return a tuple. When the logical expression is simplified, it may product the final evaluated state, i.e. true|false. In other cases, it returns the expression.

Saying that, I will drop a note about comments for all new functions, as this is a public repo, we should document thoroughly

Reply:
Oh I see now.  Right.  Thank you for the explanation.